packages/react-table/src/components/BaseTable/BaseBodyCell.tsx

```diff
@@ -0,0 +1,194 @@
+import * as React from 'react';
+import { css } from '@patternfly/react-styles';
+import styles from '@patternfly/react-styles/css/components/Table/table';
+import {
+  OnSelect,
+  selectable,
+  cellActions,
+  /* eslint-disable @typescript-eslint/no-unused-vars */
+  IAreActionsDisabled,
+  IActionsResolver,
+  expandedRow,
+  /* eslint-enable @typescript-eslint/no-unused-vars */
+  IActions,
+  IFormatterValueType,
+  collapsible,
+  IRowData,
+  IExtraData
+} from '../Table';
+import { mergeProps } from '../Table/base/merge-props';
+import {
+  DropdownDirection,
+  DropdownPosition
+} from '@patternfly/react-core/dist/js/components/Dropdown/dropdownConstants';
+
+export interface BaseBodyCellProps extends Omit<React.HTMLProps<HTMLTableDataCellElement>, 'onSelect'> {
+  /** Content rendered inside the <td> body cell */
+  children?: React.ReactNode;
+  /** Additional classes added to the <td> body cell  */
+  className?: string;
+  /** Element to render */
+  component?: React.ReactNode;
+  /**
+   * The column header the cell corresponds to.
+   * This attribute replaces table header in mobile viewport. It is rendered by ::before pseudo element.
+   */
+  dataLabel?: string;
+  /** The column index */
+  columnIndex?: number;
+  /** The row index */
+  rowIndex?: number;
+  /** Modifies cell to center its contents. */
+  textCenter?: boolean;
+  /** Style modifier to apply */
+  modifier?: 'breakWord' | 'fitContent' | 'nowrap' | 'truncate' | 'wrap';
+  /** Transforms the cell into a selectable cell - Click callback on select */
+  onSelect?: OnSelect;
+  /** Whether the checkbox is selected */
+  isSelected?: boolean;
+  /** The selectable variant */
+  selectVariant?: 'checkbox' | 'radio';
+  /** Whether all rows are selected */
+  allRowsSelected?: boolean;
+  /** Whether to disable the selection */
+  disableSelection?: boolean;
+  /** Cell actions */
+  actions?: IActions;
+  /** Whether to disable the actions */
+  actionsDisabled?: boolean;
+  /** Actions dropdown position */
+  actionsDropdownPosition?: DropdownPosition;
+  /** Actions dropdown direction */
+  actionsDropdownDirection?: DropdownDirection;
+  /** Turns the cell into an expansion toggle */
+  isExpanded?: boolean;
+  /** On toggling the expansion */
+  onCollapse?: (
+    event: React.MouseEvent,
+    rowIndex: number,
+    isOpen: boolean,
+    rowData: IRowData,
+    extraData: IExtraData
+  ) => undefined;
+  /** True to remove padding */
+  noPadding?: boolean;
```

Comment:
Can you clarify what you mean?

Reply:
When I first reviewed this I didn't realize this code is shared between `Table` and `BaseTable`. I was hoping for props that set state like `allRowsSelected` and `onSelect` to be omitted from `Base` components to keep them simpler. Then the consumer can implement setting `isExpanded` or `isSelected` themselves however they please.

This is still possible, it just makes the `Base` components more complex since they have to handle being used in the Table monolith.